Controller/Callback/Index.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace Op\Checkout\Controller\Callback;
+
+use Magento\Checkout\Model\Session;
+use Magento\Framework\App\Action\Context;
+use Magento\Quote\Model\QuoteRepository;
+use Op\Checkout\Gateway\Validator\ResponseValidator;
+use Op\Checkout\Helper\ProcessPayment;
+use Op\Checkout\Model\CheckoutException;
+use Op\Checkout\Model\TransactionSuccessException;
+use Op\Checkout\Model\ReceiptDataProvider;
+
+/**
+ * Class Index
+ */
+class Index extends \Magento\Framework\App\Action\Action
+{
+    /**
+     * @var Session
+     */
+    protected $session;
+
+    /**
+     * @var ResponseValidator
+     */
+    protected $responseValidator;
+
+    /**
+     * @var ReceiptDataProvider
+     */
+    protected $receiptDataProvider;
+
+    /**
+     * @var QuoteRepository
+     */
+    protected $quoteRepository;
+
+    /**
+     * @var ProcessPayment
+     */
+    private $processPayment;
+
+    /**
+     * Index constructor.
+     * @param Context $context
+     * @param Session $session
+     * @param ResponseValidator $responseValidator
+     * @param QuoteRepository $quoteRepository
+     * @param ReceiptDataProvider $receiptDataProvider
+     * @param ProcessPayment $processPayment
+     */
+    public function __construct(
+        Context $context,
+        Session $session,
+        ResponseValidator $responseValidator,
+        QuoteRepository $quoteRepository,
+        ReceiptDataProvider $receiptDataProvider,
```

Comment:
Do we need all of these dependencies here?

Reply:
removed